lib/boxmaker.scad

```diff
@@ -0,0 +1,176 @@
+include <flat_pack.scad>;
+
+/*
+ * A finger-jointed lasercut box maker OpenSCAD library.
+ *
+ * Inspired by http://boxdesigner.connectionlab.org/
+ */
+module box_2d(box_inner, thickness, tabs, margin=2) {
+  layout_2d(box_inner, thickness) {
+    side_a_top(box_inner, thickness, tabs);
+    side_a(box_inner, thickness, tabs);
+    side_b(box_inner, thickness, tabs);
+    side_b(box_inner, thickness, tabs);
+    side_c(box_inner, thickness, tabs);
+    side_c(box_inner, thickness, tabs);
+  }
+}
+
+// An assembled version of the box for previewing.
+module box_3d(box_inner, thickness, tabs) {
+  layout_3d(box_inner, thickness) {
+    side_a_top(box_inner, thickness, tabs);
```

Comment:
Adding some comments about which plane each of these works in might be helpful - e.g. side_a is in the XY plane

Reply:
Good idea. A good software practice is to write your functions/modules/classes, etc. such that comments like that aren't necessary. I'll rename them according to their planes instead of the arbitrary names they have.

Reply:
In my boxmaker I used face_xy face_zy and so on